### PR TITLE
Fix io error when extracing zip archive containing folders

### PR DESCRIPTION
**Pull request by t-mw**
If a binary is contained in a folder in a zip archive (i.e. `bin_path_in_archive` has a prefix), extracting will fail with an io error "No such file or directory", because `fs::File::create` in `Extract::extract_into` and `Extract::extract_path` is passed the path of the binary including the zipped parent folder, which does not exist yet.

The commit adds tests for these cases and cleans up some existing tests.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -994,84 +939,148 @@ mod tests {
...
+            "self_update_unpack_zip_src",
+            "archive.zip",
+            ArchiveKind::Zip,
+        );
+    }
+
+    #[cfg(not(feature = "archive-zip"))]
+    #[test]
+    #[ignore]
+    fn unpack_zip_file() {
```
Comment (jaemk) [inline src/lib.rs:952]: this needed or can it be deleted?

**Review by jaemk — COMMENTED** [review]
Thanks, looks good @t-mw - mind resolving the conflict before I merge?

File: src/lib.rs (hunk 1 above)
Comment (t-mw) [reply src/lib.rs:952]: This function? The other tests do the same so I just copied them, but happy to remove it.

Comment (jaemk) [reply src/lib.rs:955]: ah right, that's fine then

**Comment by jaemk** [comment]
released in 0.25.0

**Pull request merged**